### ISSUE TITLE
Infinite loop between Layout/SpaceAroundOperators and Layout/HashAlignment with EnforcedHashRocketStyle table

### ISSUE THREAD

**Issue by henrahmagix** [rubocop/rubocop#8099]
## Expected behavior

`Layout/SpaceAroundOperators` with `AllowForAlignment: true` should not complain when a hash is aligned with hashes above it.

## Actual behavior

`Layout/SpaceAroundOperators` complains about extra space, and when autocorrected it is undone by `Layout/HashAlignment` autocorrection, and an infinite loop is reported.

It is also impossible to manually fix for the same reason: fixing for `Layout/SpaceAroundOperators` then makes `Layout/HashAlignment` complain, and fixing for that takes us back to the original `Layout/SpaceAroundOperators` complaint – an endless cycle 😢 

No problems with colon-style, only rocket-style.

## Steps to reproduce the problem

- Have the following rubocop config:
```yml
Layout/SpaceAroundOperators:
  Enabled: true
  AllowForAlignment: true

Layout/HashAlignment:
  Enabled: true
  EnforcedHashRocketStyle:
    - table
    - key
```
- Create the following file:
```rb
# bug.rb in the output above
hash = {
  "one"   => 1,
  "hash"  => {

  },
  "align" => {
    "hello" => "value"
  },
  "deep"  => {
    "hash" => {
      "this is" => "deep"
    }
  }
}
```
- Run `rubocop --only Layout --fix-layout` on that file.
- See the following output:
```
$ rubocop --only Layout --fix-layout bug.rb
Inspecting 1 file
C

Offenses:

bug.rb:9:3: C: [Corrected] Layout/HashAlignment: Align the keys and values of a hash literal if they span more than one line.
  "deep" => { ...
  ^^^^^^^^^^^
bug.rb:9:11: C: [Corrected] Layout/SpaceAroundOperators: Operator => should be surrounded by a single space.
  "deep"  => {
          ^^

0 files inspected, 2 offenses detected, 2 offenses corrected
Infinite loop detected in /Users/me/Code/my_gemset/bug.rb.
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:287:in `check_for_infinite_loop'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:271:in `block in iterate_until_no_changes'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:270:in `loop'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:270:in `iterate_until_no_changes'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:241:in `do_inspection_loop'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:122:in `block in file_offenses'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:146:in `file_offense_cache'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:120:in `file_offenses'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:111:in `process_file'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:90:in `block in each_inspected_file'
...etc...
```

This also happens on the `align` hash key if you rename it to `x`, so the preceeding hash doesn't have to contain anything, it just has to have `{` `}` on separate lines:
```rb
hash = {
  "one"  => 1,
  "hash" => {

  },
  "x" => {
    "hello" => "value"
  },
  "deep" => {
    "hash" => {
      "this is" => "deep"
    }
  }
}
```
```
$ rubocop --only Layout --fix-layout bug.rb
Inspecting 1 file
C

Offenses:

bug.rb:2:3: C: [Corrected] Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
  "one"   => 1,
  ^^^^^^^^^^^^
bug.rb:3:3: C: [Corrected] Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
  "hash"  => { ...
  ^^^^^^^^^^^^
bug.rb:6:3: C: [Corrected] Layout/HashAlignment: Align the keys and values of a hash literal if they span more than one line.
  "x" => { ...
  ^^^^^^^^
bug.rb:6:10: C: [Corrected] Layout/SpaceAroundOperators: Operator => should be surrounded by a single space.
  "x"    => {
         ^^
bug.rb:9:3: C: [Corrected] Layout/HashAlignment: Align the keys of a hash literal if they span more than one line.
  "deep"  => { ...
  ^^^^^^^^^^^^
bug.rb:9:11: C: [Corrected] Layout/SpaceAroundOperators: Operator => should be surrounded by a single space.
  "deep"  => {
          ^^

0 files inspected, 6 offenses detected, 6 offenses corrected
Infinite loop detected in /Users/me/Code/my_gemset/bug.rb.
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:287:in `check_for_infinite_loop'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:271:in `block in iterate_until_no_changes'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:270:in `loop'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:270:in `iterate_until_no_changes'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:241:in `do_inspection_loop'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:122:in `block in file_offenses'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:146:in `file_offense_cache'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:120:in `file_offenses'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:111:in `process_file'
/Users/me/.rbenv/versions/2.6.6/gemsets/my_gemset/gems/rubocop-0.84.0/lib/rubocop/runner.rb:90:in `block in each_inspected_file'
...etc...
```

But no problem with colon-style hashes:
```rb
hash = {
  one: 1,
  hash: {

  },
  align: {
    hello: "value"
  },
  deep: {
    hash: {
      "this is": "deep"
    }
  }
}
```
```yml
Layout/SpaceAroundOperators:
  Enabled: true
  AllowForAlignment: true

Layout/HashAlignment:
  Enabled: true
  EnforcedColonStyle:
    - table
    - key
  EnforcedHashRocketStyle:
    - table
    - key
```
```
$ rubocop --only Layout --fix-layout bug.rb
Inspecting 1 file
.

1 file inspected, no offenses detected
```

## RuboCop version

0.84.0 (using Parser 2.7.1.3, rubocop-ast 0.0.3, running on ruby 2.6.6 x86_64-darwin19)

**Comment by henrahmagix** [comment]
I just tested this with version `0.9.2` and the latest version `1.6.1`, and this is still an issue with both.

Is it a complex one that is unlikely to be fixed?

I attempted to look into it myself, but the source code for these two Layout Cops is unfortunately beyond me at the moment – I only wrote my first linter a few weeks ago so I still have some learning =)

**Comment by kevinrobell-st** [comment]
This still seems to be an issue as of RuboCop `v1.55.1`.

**Comment by mvare007** [comment]
Still occurs as of Rubocop  `v1.63.5`. 
In my case, caused by Layout/EmptyLinesAroundAccessModifier -> Layout/EmptyLinesAroundClassBody.

**Comment by dvandersluis** [comment]
@mvare007 can you open a separate issue, with an example?
